runtime_test.go

```diff
@@ -665,6 +665,49 @@ func TestRuntime_Close_ClosesCompiledModules(t *testing.T) {
 	}
 }
 
+func TestRuntime_Close_Instantiate_ShouldErr(t *testing.T) {
+	for _, tc := range []struct {
+		name            string
+		instantiateFunc func(r Runtime) error
+	}{
+		{
+			name: "InstantiateModuleFromBinary",
+			instantiateFunc: func(r Runtime) error {
+				_, err := r.InstantiateModuleFromBinary(testCtx, binaryNamedZero)
+				return err
+			},
+		},
+		{
+			name: "CompileModule",
+			instantiateFunc: func(r Runtime) error {
+				_, err := r.CompileModule(testCtx, binaryNamedZero)
+				return err
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			engine := &mockEngine{name: "mock", cachedModules: map[*wasm.Module]struct{}{}}
+			conf := *engineLessConfig
+			conf.newEngine = func(context.Context, api.CoreFeatures, filecache.Cache) wasm.Engine { return engine }
+			r := NewRuntimeWithConfig(testCtx, &conf)
+			defer r.Close(testCtx)
+
+			// Normally compiled modules are closed when instantiated but this is never instantiated.
+			_, err := r.CompileModule(testCtx, binaryNamedZero)
+			require.NoError(t, err)
+			require.Equal(t, uint32(1), engine.CompiledModuleCount())
+
+			err = r.Close(testCtx)
+			require.NoError(t, err)
+
+			// Closing the runtime should remove the compiler cache if cache is not configured.
+			require.True(t, engine.closed)
+
+			require.Error(t, tc.instantiateFunc(r))
```

Comment:
I think we should use EqualError here, as the Error could otherwise be cryptic in a way we'd not notice